`vortex_arrays::assertions::assert_arrays_eq` forces users to install `itertools`

### Discussed in https://github.com/vortex-data/vortex/discussions/5542

<div type='discussions-op-text'>

<sup>Originally posted by **harryscholes** November 26, 2025</sup>
### Issue Description

`assert_arrays_eq!` forces users to install `itertools` inside the calling crate. The `itertools` dependency inside `vortex-arrays` does not get used because the macro splices the macro body into the calling code. Adding `itertools` to the calling crate's `Cargo.toml` works.

### Expected Behavior

`assert_arrays_eq!` should just work.

### Actual Behavior

```
error[E0433]: failed to resolve: use of unresolved module or unlinked crate `itertools`
  --> src/main.rs:58:5
   |
58 |     assert_arrays_eq!(&a, &a);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^ use of unresolved module or unlinked crate `itertools`
   |
   = help: if you wanted to use a crate named `itertools`, use `cargo add itertools` to add it to your `Cargo.toml`
   = note: this error originates in the macro `assert_arrays_eq` (in Nightly builds, run with -Z macro-backtrace for more info)
```

### Reproduction Steps

```rs
use vortex::{arrays::PrimitiveArray, buffer::buffer, validity::Validity};
use vortex_array::assert_arrays_eq;

fn main() {
    let a = PrimitiveArray::new(buffer![1u64, 2, 3], Validity::NonNullable);
    assert_arrays_eq!(&a, &a);
}
```

### OS Version Information

macOS 15.6.1

### I acknowledge that:

- [X] I have searched the Vortex repository (both open and closed Discussions and Issues) and confirm this is not a duplicate of an existing issue or discussion.
- [X] I have checked the "Preview" tab on all text fields to ensure that everything looks right, and have wrapped all configuration and code in code blocks with a group of three backticks (` ``` `) on separate lines.</div>

--- a/vortex-array/src/arrays/assertions.rs
+++ b/vortex-array/src/arrays/assertions.rs
@@ -1,6 +1,14 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-FileCopyrightText: Copyright the Vortex contributors
 
+use std::fmt::Display;
+
+use itertools::Itertools;
+
+pub fn format_indices<I: IntoIterator<Item = usize>>(indices: I) -> impl Display {
+    indices.into_iter().format(",")
+}
+
 #[macro_export]
 macro_rules! assert_arrays_eq {
     ($left:expr, $right:expr) => {{
@@ -32,7 +40,7 @@ macro_rules! assert_arrays_eq {
         if mismatched_indices.len() != 0 {
             panic!(
                 "assertion left == right failed: arrays do not match at indices: {}.\n  left: {}\n right: {}",
-                itertools::Itertools::format(mismatched_indices.into_iter(), ", "),
+                $crate::arrays::format_indices(mismatched_indices),
                 left.display_values(),
                 right.display_values()
             )

--- a/vortex-array/src/arrays/mod.rs
+++ b/vortex-array/src/arrays/mod.rs
@@ -6,6 +6,9 @@
 #[cfg(any(test, feature = "test-harness"))]
 mod assertions;
 
+#[cfg(any(test, feature = "test-harness"))]
+pub use assertions::format_indices;
+
 #[cfg(test)]
 mod validation_tests;
 

